presentation/virtualKeyboard.brs

```diff
@@ -3,41 +3,79 @@ REM @title                   BrightSign Virtual Keyboard
 REM @author                  Lee Dydo
 REM @company                 BrightSign LLC.
 REM @date-created            06/06/2016
-REM @date-last-modified      07/05/2016
+REM @date-last-modified      06/07/2016
 REM
+
+' This is an example plugin script for the BrightSign Virtual Keyboard, used in your BA:Con presentation
+' For information on how to use this script, please see the readme in the github repo or this script's comments
+
 Function virtualKeyboard_Initialize(msgPort As Object, userVariables As Object, bsp As Object)
   return {
     objectName :      "virtualKeyboard_object",
-    version:          1.02,
     msgPort :         msgPort,
     userVariables :   userVariables,
     bsp :             bsp,
     rectangle :       invalid,
     virtualKeyboard : invalid,
     ProcessEvent :    Function (event As Object)
       'Keyboard needs to be initialized after the target roHTMLWidget, in order to be rendered on top
-	  if m.virtualKeyboard = invalid
-	    vm = m.bsp.videomode
-		width = vm.getresx()
-		height = vm.getresy()
+      if m.virtualKeyboard = invalid
         keyboardPackage = createObject("roBrightPackage", m.bsp.assetPoolFiles.getPoolFilePath("keyboard.zip"))
         if type (keyboardPackage) = "roBrightPackage"
           createDirectory("virtualKeyboard")
           keyboardPackage.unpack("virtualKeyboard/")
         endif
-		rectangle = createObject("rorectangle", 0, 0, width, height)
-        virtualKeyboard = createObject("roVirtualKeyboard", rectangle)
-        if m.bsp.sign.monitorOrientation = "portrait" then
-		  virtualKeyboard.setTransform("rot90")
-		else if m.bsp.sign.monitorOrientation = "portraitbottomonright" then
-		  virtualKeyboard.setTransform("rot270")
-		endif
-		virtualKeyboard.setResource("file:///virtualKeyboard/bsvirtualkb.html")
-        virtualKeyboard.setPort(m.msgPort)
-		m.virtualKeyboard = virtualKeyboard
-		m.rectangle = rectangle
-		m.virtualKeyboard.hide()
-	  endif
+
+        ' Need to do some math to get the correct rectangle for the virtual keyboard
+        ' This is based on the proportions designed around native 1080p resolution
+        vidmode = createObject("roVideoMode")
+        width = vidmode.getResX()
+        height = vidmode.getResY()
+
+        ' rectX and rectY are the dimensions of the keyboard
+        rectX = Cint(width*.3125) ' Using percentage adjustments based on the resolution allows for correct sizing regardless of res
```

Comment:
Is it the case that _rectX_ and _rectY_ are the width, height, or x and y coordinate of the keyboard? The variable or comment may need updating.

Reply:
rectx/recty are width/height and startx/starty are the coordinates. Will update to make more understandable